api/scripts/daily-digest.ts

```diff
@@ -0,0 +1,125 @@
+import { SMTPClient } from "emailjs";
+import { createEmailWithTemplate } from "../emails/htmlEmailUtils.js";
+import config from "@config";
+import type { MonitoringParams } from "../api/V1/monitoringPage.js";
+import { calculateMonitoringPageCriteria } from "../api/V1/monitoringPage.js";
+import { generateVisits } from "../api/V1/monitoringVisit.js";
+import modelsInit from "@models/index.js";
+import { RecordingType } from "@typedefs/api/consts.js";
+
+(async () => {
+  try {
+    const models = await modelsInit();
+    const users = await models.User.findAll({where: {
+      'settings.emailNotifications.dailyDigest': true
+    }});
+    
+    const templateFilename = "daily-digest.html";
+    const client = new SMTPClient(config.smtpDetails);
+    // const client = new SMTPClient({
+    //   user: "",
+    //   password: "",
+    //   host: "smtp.gmail.com",
+    //   ssl: true,
+    // });
+
+    users.forEach(async user => {
+      const requestUser = user;
+      const groups = await models.GroupUsers.findAll({where: {
+        'UserId': user.id
+      }});
+
+      groups.forEach(async group => {
+        const recordingData = {};
+        const name = await models.Group.findOne({where: {
```

Comment:
You've already got the group object, so you shouldn't need to query the DB again here just to get its name – you already have that on the `group` variable.

Reply:
The Groups table contains the groupName but does not have the associated users in the group. The GroupUsers contains the associated users but not the groupName. What I'm doing is finding all the groups that the specific user is part of from the GroupUsers table, then from the groups the user is a part of I am getting the the group ID and using that to find the name of the group. Since I need both the group name and the group users I think both queries are necessary as neither table contains both pieces of information.

Reply:
Ah, right, so what you'd usually do is do that in one query, using the include param i.e: 
```
include: [
      {
        model: models.Group,
        attributes: ["id", "groupName"],
        required: true,
}],
```